### PR TITLE
Add config validation

### LINKED ISSUE #33: Validate elasticsearch timeout value
The elasticsearch timeout value from pelias config is sent to the elasticsearch-js module and eventually [directly to setTimeout](https://github.com/elastic/elasticsearch-js/blob/master/src/lib/transport.js#L363-L366), with no validation along the way. As it happens, this means invalid values are often parsed as zero milisecond timeouts, which are of course too short to be meaningful. When this happens the importers return confusing error messages like this one:

```
2016-11-14T21:08:20.110Z - error: [dbclient] invalid resp from es bulk index operation
2016-11-14T21:08:20.110Z - info: [dbclient] retrying batch [500]
2016-11-14T21:08:20.114Z - error: [dbclient] esclient error Error: Request Timeout after 120_000ms
    at /home/julian/repos/pelias/whosonfirst/node_modules/elasticsearch/src/lib/transport.js:336:15
    at Timeout.<anonymous> (/home/julian/repos/pelias/whosonfirst/node_modules/elasticsearch/src/lib/transport.js:365:7)
    at tryOnTimeout (timers.js:228:11)
    at Timer.listOnTimeout (timers.js:202:5)
```

It would also make sense upstream in the Elasticsearch module, but we should do some validation of our own and enforce a reasonable minimum value.

### PR DESCRIPTION
**Pull request by trescube**
Fixes #33

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,2 +1,6 @@
+const config = require('pelias-config');
 
-module.exports = require('./src/sink');
\ No newline at end of file
+module.exports = (function() {
+  require('./src/configValidation').validate(config.generate());
```
Comment (orangejulius) [inline index.js]: tricky. does it work if you move line 4 out on its own? If so you can get rid of the [IIFE](http://benalman.com/news/2010/11/immediately-invoked-function-expression/) which would be nice.

**Comment by orangejulius** [comment]
Can we easily fix #33 here as well by adding a requirement that the `config.esclient.requestTimeout` value is an integer? Or is there a better place for that?

File: index.js (hunk 1 above)
Comment (trescube) [inline index.js]: Yep, works outside, moved.

**Comment by trescube** [comment]
Yep, I didn't want to get into validating the object passed to elasticsearch but this seems like a reasonable ask.

**Comment by trescube** [comment]
Fixes #33

**Pull request merged**